packages/image_picker/example/lib/main.dart

```diff
@@ -32,53 +33,452 @@ class MyHomePage extends StatefulWidget {
 }
 
 class _MyHomePageState extends State<MyHomePage> {
-  Future<File> _imageFile;
+  Future<File> _mediaFile;
+  bool isVideo = false;
+  VideoPlayerController _controller;
+  VoidCallback listener;
 
   void _onImageButtonPressed(ImageSource source) {
     setState(() {
-      _imageFile = ImagePicker.pickImage(source: source);
+      if (_controller != null) {
+        _controller.setVolume(0.0);
+        _controller.removeListener(listener);
+      }
+      if (isVideo) {
+        _mediaFile = ImagePicker.pickVideo(source: source).then((File _file) {
```

Comment:
Several issues here:
* Please don't use underscores for local variables.
* This fails, if the user cancels the selection: `_file` will be `null`, and you cannot construct a `VideoPlayerController` from that.
* Even if the user selects a video, `_mediaFile` is going to complete with `null`, because we don't return the `_file` from the lambda given to `then`.
* The code has become hard to understand, because we don't use `_mediaFile` in the video case. This should be refactored for clarity.